runtime/stdlib/contracts/test.cdc

```diff
@@ -0,0 +1,236 @@
+            return self.backend.createAccount()
+        }
+
+        /// Add a transaction to the current block.
+        ///
+        pub fun addTransaction(_ tx: Transaction) {
+            self.backend.addTransaction(tx)
+        }
+
+        /// Executes the next transaction in the block, if any.
+        /// Returns the result of the transaction, or nil if no transaction was scheduled.
+        ///
+        pub fun executeNextTransaction(): TransactionResult? {
+            return self.backend.executeNextTransaction()
+        }
+
+        /// Commit the current block.
+        /// Committing will fail if there are un-executed transactions in the block.
+        ///
+        pub fun commitBlock() {
+            self.backend.commitBlock()
+        }
+
+        /// Executes a given transaction and commit the current block.
+        ///
+        pub fun executeTransaction(_ transaction: Transaction): TransactionResult {
+            self.addTransaction(transaction)
+            let txResult = self.executeNextTransaction()!
+            self.commitBlock()
+            return txResult
+        }
+
+        /// Executes a given set of transactions and commit the current block.
+        ///
+        pub fun executeTransactions(_ transactions: [Transaction]): [TransactionResult] {
+            for tx in transactions {
+                self.addTransaction(tx)
+            }
+
+            var results: [TransactionResult] = []
+            for tx in transactions {
+                let txResult = self.executeNextTransaction()!
+                results.append(txResult)
+            }
+
+            self.commitBlock()
+            return results
+        }
+
+        /// Deploys a given contract, and initilizes it with the arguments.
+        ///
+        pub fun deployContract(
+            name: String,
+            code: String,
+            account: Account,
+            arguments: [AnyStruct]
+        ): Error? {
+            return self.backend.deployContract(
+                name: name,
+                code: code,
+                account: account,
+                arguments: arguments
+            )
+        }
+
+        pub fun useConfiguration(_ configuration: Configuration) {
+            self.backend.useConfiguration(configuration)
+        }
+    }
+
+    pub struct Matcher {
+
+        pub let test: ((AnyStruct): Bool)
+
+        pub init(test: ((AnyStruct): Bool)) {
+            self.test = test
+        }
+
+        /// Combine this matcher with the given matcher.
+        /// Returns a new matcher that succeeds if this and the given matcher succeed.
+        ///
+        pub fun and(_ other: Matcher): Matcher {
+            return Matcher(test: fun (value: AnyStruct): Bool {
+                return self.test(value) && other.test(value)
+            })
+        }
+
+        /// Combine this matcher with the given matcher.
+        /// Returns a new matcher that succeeds if this and the given matcher succeed.
```

Comment:
```suggestion
        /// Returns a new matcher that succeeds if this or the given matcher succeed.
```
Probably also worth documenting whether or not this is short circuiting, and on which matcher